docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,211 @@
+  - Admin marks <Training Task> as complete.
+
+## Tracking participation in interactive work.
+
+* [admin reports successful completion of interactive work]: such as discussion session or lesson demo.
+  - Admin views {training progress} for trainee.
+  - Admin selects <Training Task>.
+  - Admin:
+    - Enters date of interactive event.
+    - Adds comments (if any).
+  - Admin indicates whether <Training Task> is now complete.
+
+## Completing training work.
+
+* [admin changes completion status of work for trainee]
+  - Admin looks up <Training Progress> associated with trainee.
+  - Admin changes "completed" status of work item(s).
+  => [training automatically completed]
+
+* [admin finds work to review]
+  - Admin views {training progress display}.
+  - Admin selects <Training Progress>.
+  - Admin => [admin changes completion status of work for trainee].
+
+## Completing training.
+
+* [training automatically completed]: triggered by changes in <Training Task>s.
+  - AMY records change to <Training Task> and all <Training Task>s for <Person> now complete.
+  - AMY displays {badge creation form}.
+  - Admin confirms creation of badge.
+  - AMY moves <Training Progress> to <Past Training Progress List>.
+
+## Ending training.
+
+* [admin ends training]: because trainee has gone silent, failed a work item, etc.
+  - Admin views {training progress display}.
+  - Admin selects <Training Progress>.
+  - Admin marks <Training Progress> as "unsuccessful".
+  - AMY moves <Training Progress> to <Past Training Progress List>.
+
+* [admin ends training course]: because it has timed out.
+  - Admin selects training <Event>.
+  - Admin marks <Event> as "stalled".
+  - AMY marks all associated <Training Progress>es as "unsuccessful".
+  - AMY moves <Training Progress>es to <Past Training Progress List>.
+
+## Creating certificates.
+
+* [admin creates certificate]: for trainers who don't yet have them.
+  - Admin goes to {certificate creation form}.
+  - Admin selects <Person>s who:
+    - Have an instructor badge.
+    - Do not yet have a PDF certificate.
+  - Admin adjusts information in form.
+    - Changes name of instructor(s) awarding certificates.
+    - Changes date of award.
+  - AMY generates PDF certificates.
+  - User views/downloads certificate on {certificate display}.
+
+## Questions
+
+- What to do if Admin deletes <Task> from training <Event> directly rather than via [admin removes people from training event]?
+- What workflow *should* we use for marking training events as closed/stalled?
+- Do we only allow feedback loop/multiple tries for interactive work (discussions and demos), but *not* for written work?  That's how the workflows are written now...
+- Can we support group applications by providing one extra free-text field in the application called "Group Name", and asking people who want to co-apply to fill it in with the same value?  We can then sort/filter/group by that value to find people in the same group.
```

Comment:
That's a great idea! The only question is if a group of people will want to apply if every person in the group have to fill the form, instead of delegating one person to fill one group application. And one minor issue: in that case we don't want `reason` field in every application, only one for one group.